tests/server/test_params.py

```diff
@@ -0,0 +1,226 @@
+"""Unit tests for granular sensor authentication in api.php."""
+
+# standard library
+import unittest
+import base64
+
+# from flask.testing import FlaskClient
+from delphi.epidata.server._common import app
+from delphi.epidata.server._params import (
+    parse_geo_arg,
+    parse_source_signal_arg,
+    parse_time_arg,
+    parse_day_value,
+    parse_week_value,
+    GeoPair,
+    TimePair,
+    SourceSignalPair,
+)
+from delphi.epidata.server._exceptions import (
+    ValidationFailedException,
+)
+
+# py3tester coverage target
+__test_target__ = "delphi.epidata.server._params"
+
+
+class UnitTests(unittest.TestCase):
+    """Basic unit tests."""
+
+    # app: FlaskClient
+
+    def setUp(self):
+        app.config["TESTING"] = True
+        app.config["WTF_CSRF_ENABLED"] = False
+        app.config["DEBUG"] = False
+
+    def test_parse_geo_arg(self):
+        with self.subTest("empty"):
+            with app.test_request_context("/"):
+                self.assertEqual(parse_geo_arg(), [])
+        with self.subTest("single"):
+            with app.test_request_context("/?geo=state:*"):
+                self.assertEqual(parse_geo_arg(), [GeoPair("state", True)])
+            with app.test_request_context("/?geo=state:AK"):
+                self.assertEqual(parse_geo_arg(), [GeoPair("state", ["AK"])])
```

Comment:
CSV ingestion automatically puts all geo values through tolower() before storing them in the database; do we want to automatically do the same for query parameters?

Reply:
we can. So far they end up in an SQL command which is usually case insensitive

Reply:
while TIL we're currently using case-insensitive collation for all epidata tables, some of the MariaDB storage engines default to a case-sensitive collation. I don't want to set us up for surprises down the line if we ever have to rebuild the system on another engine

Reply:
I changed the code already to convert it to lowercase